src/foundry/common/utils/bitmask.d.mts

```diff
@@ -0,0 +1,147 @@
+export default class BitMask extends Number {
+  /**
+   * Create a new BitMask instance.
+   * @param states    - An object containing valid states and their corresponding initial boolean values (default is null).
+   */
+  constructor(states?: Record<string, boolean> | null);
+
+  /**
+   * The real value behind the bitmask instance.
+   */
+  #value: number;
+
+  /**
+   * The structure of valid states and their associated values.
+   */
+  #validStates: Map<string, number>;
+
+  /**
+   * The enum associated with this structure.
+   * @readonly
+   */
+  states: Record<string, string>;
+
+  /**
+   * Generates the valid states and their associated values.
+   * @param states The structure defining the valid states and their associated values. Defaults to null.
+   */
+  #generateValidStates(states?: Record<string, boolean> | null): void;
+
+  /**
+   * Generates an enum based on the provided valid states.
+   */
+  #generateEnum(): void;
+
+  /**
+   * Calculate the default value of the bitmask based on the initial states
+   * @param initialStates  - The structure defining the valid states and their associated values. Defaults to {}
+   * @returns           The bitmask value
+   */
+  #computeValue(initialStates?: Record<string, boolean>): number;
+
+  /**
+   * Checks a state and throws an error if it doesn't exist.
+   * @param state     - Name of the state to check.
+   */
+  #checkState(state: string): void;
+
+  /**
+   * True if this bitmask is empty (no active states).
+   */
+  get isEmpty(): boolean;
+
+  /**
+   * Check if a specific state is active.
+   * @param state   - The state to check.
+   * @returns True if the state is active, false otherwise.
+   */
+  hasState(state: string): boolean;
+
+  /* -------------------------------------------- */
```

Comment:
We don't need these comments in our source. I know we have some elsewhere but we'll delete them, eventually. It's just not worth the churn to have them.

Reply:
none of the comments?  or just the /********/ ones?

Reply:
The /***/ ones

Reply:
Do you want me to remove them throughout the project?  That's definitely in my skillset :)

Reply:
Just handling the ones in the PR is fine, we'll address others as we fix up their files